client/blocks/importer/wordpress/upgrade-plan/upgrade-plan-details.tsx

```diff
@@ -1,24 +1,220 @@
+					onClick={ onAnnualPlanClick }
+				>
+					{ __( 'Pay annually' ) }
+				</Button>
+			</ButtonGroup>
+		</div>
+	);
+};
+
+const Badge = styled.div< { isForIntroOffer?: boolean; isHidden?: boolean } >`
+	text-align: center;
+	white-space: nowrap;
+	font-size: 0.75rem;
+	line-height: 1.25rem;
+	border-radius: 4px;
+	height: 21px;
+	display: inline-block;
+	width: fit-content;
+	letter-spacing: 0.2px;
+	font-weight: 500;
+	text-align: center;
+	padding: 0 12px;
+	background-color: #e6d1ff;
+	color: #350e63;
+	text-transform: none;
+`;
+
+const PricesGroup = styled.div`
+	justify-content: flex-end;
+	display: flex;
+	flex-direction: row-reverse;
+	align-items: flex-end;
+	gap: 4px;
+`;
+
+interface PriceWithIntroductoryOfferProps {
+	originalMonthlyPrice: number;
+	introOfferMonthlyPrice: number;
+	currencyCode?: string;
+}
+
+const PriceWithIntroductoryOffer = ( props: PriceWithIntroductoryOfferProps ) => {
+	const { originalMonthlyPrice, introOfferMonthlyPrice, currencyCode } = props;
+	const translate = useTranslate();
+
+	return (
+		<>
+			<Badge>{ translate( 'One time offer' ) }</Badge>
+			<PricesGroup>
+				<PlanPrice
+					rawPrice={ originalMonthlyPrice }
+					currencyCode={ currencyCode }
+					original
+					isSmallestUnit
+				/>
+				<PlanPrice rawPrice={ introOfferMonthlyPrice } currencyCode={ currencyCode } />
+			</PricesGroup>
+		</>
+	);
+};
+
+interface PlanPriceOfferProps {
+	plan?: Plan;
+	currencyCode?: string;
+	originalMonthlyPrice?: number;
+	introOfferMonthlyPrice?: number;
+	formattedOriginalPrice?: string;
+	formattedIntroOfferPrice?: string;
+	introOfferEnabled: boolean;
+	isIntroOfferComplete: boolean;
+}
+
+const PlanPriceOffer = ( props: PlanPriceOfferProps ) => {
+	const translate = useTranslate();
+
+	const {
+		plan,
+		formattedOriginalPrice,
+		formattedIntroOfferPrice,
+		introOfferEnabled,
+		isIntroOfferComplete,
+		introOfferMonthlyPrice,
+		originalMonthlyPrice,
+		currencyCode,
+	} = props;
+
+	const showIntroOffer =
+		introOfferEnabled && ! isIntroOfferComplete && introOfferMonthlyPrice && originalMonthlyPrice;
+
+	const price = showIntroOffer ? (
+		<PriceWithIntroductoryOffer
+			originalMonthlyPrice={ originalMonthlyPrice }
+			introOfferMonthlyPrice={ introOfferMonthlyPrice }
+			currencyCode={ currencyCode }
+		/>
+	) : (
+		<PlanPrice rawPrice={ originalMonthlyPrice } currencyCode={ currencyCode } isSmallestUnit />
+	);
+
+	if ( plan && showIntroOffer && formattedIntroOfferPrice ) {
+		plan.getBillingTimeFrame = () =>
+			translate(
+				'per month, %(discounted)s billed annually for the first year, %(original)s atferwards',
+				{
+					args: {
+						discounted: formattedIntroOfferPrice,
+						original: formattedOriginalPrice || '',
+					},
+				}
+			);
+	}
+
+	return (
+		<div className="import__upgrade-plan-price">
+			{ price }
+			<span className="plan-time-frame">
+				<small>{ plan?.getBillingTimeFrame() }</small>
+			</span>
+		</div>
+	);
+};
```

Comment:
At a basic level, this component is pretty straightforward, but the code above is pretty complex. Might it be worth refactoring the core logic to build the `price` and `billingTimeFrame` functions into a local hook/function, as a way of simplifying the code? As I see it (without looking deeply), that code could return early when we're not showing the introductory offer, and simply return `<PlanPrice ... />` and `plan?.getBillingTimeFrame()`. And if we are showing the introductory offer, we can return the more complex component and the custom string for the billing time frame.